src/main/java/com/acme/edu/LoggerController.java

```diff
@@ -0,0 +1,56 @@
+package com.acme.edu;
+
+import com.acme.edu.command.Command;
+import com.acme.edu.saver.Saver;
+
+import java.util.Map;
+import java.util.HashMap;
+
+public class LoggerController {
+    private static LoggerController theOnlyLogger = new LoggerController();
+    private Map<String, Boolean> commandParameters = new HashMap<>();
+    private Command previousCommand;
+    private Saver saver;
+
+    private LoggerController() {
+        resetCommandParameters();
+    }
+
+    public static LoggerController getTheOnlyLogger() {
+        return theOnlyLogger;
+    }
+
+    public void send(Command command) {
+        String result = command.process(commandParameters, previousCommand);
+        previousCommand = command;
+        saver.save(result);
+    }
+
+    public void setSaver(Saver saver) {
+        this.saver = saver;
+    }
+
+    public void setAccumulateCommandParameter(boolean accumulate) {
+        commandParameters.put("accumulate", accumulate);
```

Comment:
Looks like you design two parallel type systems – regular with command classes and encapsulated behaviour and parallel - with string constants and switching over them. (Sometimes it's called meta-type-system).
But is it really needed? For now if I'd like to add new command responsibility like accumulation - e.g. command bufferization, how much will it cost? How many code places should I add?
Maybe I can encapsulate all behaviour in classic Commands without string flags?